lte/gateway/python/integ_tests/s1aptests/test_attach_service_with_multi_pdns_and_bearers.py

```diff
@@ -0,0 +1,333 @@
+            "********************** Adding dedicated bearer to magma.ipv4"
+            " PDN"
+        )
+        print(
+            "********************** Sending RAR for IMSI",
+            "".join([str(i) for i in req.imsi]),
+        )
+        self._sessionManager_util.create_ReAuthRequest(
+            "IMSI" + "".join([str(i) for i in req.imsi]),
+            policy_id1,
+            flow_list1,
+            qos1,
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_ACT_DED_BER_REQ.value
+        )
+        act_ded_ber_req_oai_apn = response.cast(
+            s1ap_types.UeActDedBearCtxtReq_t
+        )
+        self._s1ap_wrapper.sendActDedicatedBearerAccept(
+            req.ue_id, act_ded_ber_req_oai_apn.bearerId
+        )
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Send PDN Connectivity Request
+        apn = "ims"
+        self._s1ap_wrapper.sendPdnConnectivityReq(ue_id, apn)
+        # Receive PDN CONN RSP/Activate default EPS bearer context request
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_PDN_CONN_RSP_IND.value
+        )
+        act_def_bearer_req = response.cast(s1ap_types.uePdnConRsp_t)
+        addr = act_def_bearer_req.m.pdnInfo.pAddr.addrInfo
+        sec_ip = ipaddress.ip_address(bytes(addr[:4]))
+        ips.append(sec_ip)
+
+        print(
+            "********************** Sending Activate default EPS bearer "
+            "context accept for UE id ",
+            ue_id,
+        )
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Add dedicated bearer to 2nd PDN
+        print("********************** Adding dedicated bearer to ims PDN")
+        print(
+            "********************** Sending RAR for IMSI",
+            "".join([str(i) for i in req.imsi]),
+        )
+        self._sessionManager_util.create_ReAuthRequest(
+            "IMSI" + "".join([str(i) for i in req.imsi]),
+            policy_id2,
+            flow_list2,
+            qos2,
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_ACT_DED_BER_REQ.value
+        )
+        act_ded_ber_req_ims_apn = response.cast(
+            s1ap_types.UeActDedBearCtxtReq_t
+        )
+        self._s1ap_wrapper.sendActDedicatedBearerAccept(
+            req.ue_id, act_ded_ber_req_ims_apn.bearerId
+        )
+        print(
+            "************* Added dedicated bearer",
+            act_ded_ber_req_ims_apn.bearerId,
+        )
+
+        flowRules = {
+            "num_ul_flows": ul_flows,
+            "num_dl_flows_default": dl_flows_default,
+            "num_dl_flows_sec_pdn": dl_flows_sec_pdn,
+            "ips_to_be_matched": ips,
+            "default_ip_addr": default_ip,
+        }
```

Comment:
This is not generic to be useful in other test cases. It is better to pass the list of flows to be verified for each ue ip address. E.g., pass a dictionary with key ue_ip and values flow_list.